components/support/rs_client/src/http_client.rs

```diff
@@ -76,14 +92,15 @@ impl Client {
     }
 
     fn ensure_no_backoff(&self) -> Result<()> {
+        let current_state = self.remote_state.lock().unwrap().clone();
         if let RemoteState::Backoff {
             observed_at,
             duration,
-        } = self.remote_state.get()
+        } = current_state
         {
             let elapsed_time = observed_at.elapsed();
             if elapsed_time >= duration {
-                self.remote_state.replace(RemoteState::Ok);
+                *self.remote_state.lock().unwrap() = RemoteState::Ok;
```

Comment:
If we do the above comment, we should just assign the value to `*current_state` instead of relocking the mutex

Reply:
Since we're already going through the overhead of the lock in `ensure_no_backoff`, it might be worth it to only lock once at the beginning of the function!

Reply:
yeah - I agree! Alternatively, can the caller lock it and pass it as a mutable ref? I imagine the caller is going to lock it for itself after this succeeds anyway.